src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1326,6 +1327,53 @@ macro_rules! coverage_test_alias {
     };
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Ord, PartialOrd)]
+pub struct RunMakeSupport {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RunMakeSupport {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.never()
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        let compiler = run.builder.compiler(run.builder.top_stage, run.build_triple());
+        run.builder.ensure(RunMakeSupport { compiler, target: run.build_triple() });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> PathBuf {
+        builder.ensure(compile::Std::new(self.compiler, self.target));
+
+        let cargo = tool::prepare_tool_cargo(
+            builder,
+            self.compiler,
+            Mode::ToolStd,
+            self.target,
+            "build",
+            "src/tools/run-make-support",
+            SourceType::InTree,
+            &[],
+        );
+
+        let mut cargo = Command::from(cargo);
+        cargo.env("RUSTFLAGS", "-C prefer-dynamic");
+        builder.run(&mut cargo);
+
+        let lib_name = shared_lib_name("run_make_support", &self.target.to_string());
+        let lib = builder.tools_dir(self.compiler).join(&lib_name);
+
+        let cargo_out =
+            builder.cargo_out(self.compiler, Mode::ToolStd, self.target).join(&lib_name);
+        builder.copy(&cargo_out, &lib);
+        lib
```

Comment:
I believe something here caused the build failure on Windows(?), but I need to think about what failed...

Reply:
It's the `shared_lib_name` that's wrong. On Windows, there definitely isn't a `lib` prefix for the dylib, BUT I need to find out if MSVC and MingW targets have different naming conventions for dylibs...

Reply:
I'm hoping it's as simple as Windows targets just having no `lib` prefix, i.e. `{name}.dll`...

Reply:
Indeed. `dll_prefix` in the windows target specs is empty.